include/gsl/string_span

```diff
@@ -312,83 +316,84 @@ public:
 
 private:
-    static impl_type remove_z(pointer const& sz, std::ptrdiff_t max)
+    static impl_type remove_z(pointer const& sz, std::size_t max)
     {
         return {sz, details::string_length(sz, max)};
     }
 
     template <std::size_t N>
     static impl_type remove_z(element_type (&sz)[N])
     {
-        return remove_z(&sz[0], narrow_cast<std::ptrdiff_t>(N));
+        return remove_z(&sz[0], N);
     }
 
     impl_type span_;
 };
 
-template <std::ptrdiff_t Extent = dynamic_extent>
+template <std::size_t Extent = dynamic_extent>
 using string_span = basic_string_span<char, Extent>;
 
-template <std::ptrdiff_t Extent = dynamic_extent>
+template <std::size_t Extent = dynamic_extent>
 using cstring_span = basic_string_span<const char, Extent>;
 
-template <std::ptrdiff_t Extent = dynamic_extent>
+template <std::size_t Extent = dynamic_extent>
 using wstring_span = basic_string_span<wchar_t, Extent>;
 
-template <std::ptrdiff_t Extent = dynamic_extent>
+template <std::size_t Extent = dynamic_extent>
 using cwstring_span = basic_string_span<const wchar_t, Extent>;
 
-template <std::ptrdiff_t Extent = dynamic_extent>
+template <std::size_t Extent = dynamic_extent>
 using u16string_span = basic_string_span<char16_t, Extent>;
 
-template <std::ptrdiff_t Extent = dynamic_extent>
+template <std::size_t Extent = dynamic_extent>
 using cu16string_span = basic_string_span<const char16_t, Extent>;
 
-template <std::ptrdiff_t Extent = dynamic_extent>
+template <std::size_t Extent = dynamic_extent>
 using u32string_span = basic_string_span<char32_t, Extent>;
 
-template <std::ptrdiff_t Extent = dynamic_extent>
+template <std::size_t Extent = dynamic_extent>
 using cu32string_span = basic_string_span<const char32_t, Extent>;
 
 //
 // to_string() allow (explicit) conversions from string_span to string
 //
 
-template <typename CharT, std::ptrdiff_t Extent>
+template <typename CharT, std::size_t Extent>
 std::basic_string<typename std::remove_const<CharT>::type>
 to_string(basic_string_span<CharT, Extent> view)
 {
     return {view.data(), narrow_cast<std::size_t>(view.length())};
 }
 
 template <typename CharT, typename Traits = typename std::char_traits<CharT>,
-          typename Allocator = std::allocator<CharT>, typename gCharT, std::ptrdiff_t Extent>
+          typename Allocator = std::allocator<CharT>, typename gCharT, std::size_t Extent>
 std::basic_string<CharT, Traits, Allocator> to_basic_string(basic_string_span<gCharT, Extent> view)
 {
     return {view.data(), narrow_cast<std::size_t>(view.length())};
 }
 
-template <class ElementType, std::ptrdiff_t Extent>
+template <class ElementType, std::size_t Extent>
 basic_string_span<const byte, details::calculate_byte_size<ElementType, Extent>::value>
 as_bytes(basic_string_span<ElementType, Extent> s) noexcept
 {
-    GSL_SUPPRESS(type.1) // NO-FORMAT: attribute
+    GSL_SUPPRESS(type .1) // NO-FORMAT: attribute
```

Comment:
Is a space acceptable here? Occurs below.

Reply:
this got hit with my clang-format pass. Thanks for pointing that out. Fixing.